### ISSUE TITLE
Support Symfony 6

### ISSUE THREAD

**Issue by tacman** [webfactory/WebfactoryNewsletterRegistrationBundle#5]
Do you plan to support Symfony 6?  Also, what do you think about bumping the version and dropping support for any unsupported Symfony versions?  That is, only support Symfony 4.4, 5.4 and ^6?  (Symfony will end support for 5.3 in 10 days).

**Comment by MalteWunsch** [comment]
Hi tacman!

Unfortunately, we don't have plans to work on this bundle in the near future. But if you'd like to add Symfony 6 support, we'd be happy to get a contribution!

Personally, I'm fine with supporting only `4.4.* | 5.4.* | ^6.0` - do you agree, @mpdude ?

**Comment by mpdude** [comment]
Well, generally yes, but...

I would not drop support for an older (EOL) Symfony release just for the sake of it. If we can support it without additional effort, I think we should do so.

I know that other (reputable) people in the Symfony ecosystem take a different stance on that and are eager dropping support for older releases to "force" users to upgrade in a timely manner. I think everyone using and working with software in a responsible and professional manner will do their best to keep up to date, but we don't need to push or "punish" them.

Having said that, when implementing a new feature or fixing a bug is not easily possible without leaving behind support for EOL Symfony (or Twig, or whatever) versions (because, for example, we'd need different code paths, adapter classes and the like to address non-BC changes), then I'd stick to the versions supported at that time.

Does that make sense?

**Comment by algo26-matthias** [comment]
Hi fellas,

first, congrats to this really well-built piece of software! I am currently evaluating whether to write something like this for a customer project myself or using a third-party package. Yours looks the mos promising for my use case so far.
Do you plan to add support for Symfony 6 in the near future? I mean, it's over a year since this question first came up. :-)

Best wishes
Matthias

**Comment by tacman** [comment]
What about a new release?  Anyone still using older version can use ^1.0, anyone with Symfony 6 (or 7, which is now only 6 months away) can use ^2.0.

Alas, there are no releases yet here, and managing the release process is something that sounds good but I still struggle with in my own code.  Nonetheless, if someone who knew the process better could release the current code as version 1, that would be a start.  Alas, anyone who is using the bundle in their code is locked to dev-master, so it would break on composer update (but would work with version 1).

Symfony 6 and PHP 8.1 offer so many great DX improvements that it's worth the effort.  Symfony 6.1 allows for MUCH simpler bundle configuration.  And tools like rector make moving to PHP 8 pretty easy.

**Comment by MalteWunsch** [comment]
Thanks you Matthias!

While we would like to support Symfony 6 with this bundle (and benefit from all of the nice DX improvements tacman speaks about!), we do not have a schedule for that. Usually, we'd work on that when one of our clients using this bundle updates to a new Symfony version in one of their main projects. This may take a long while, so please don't rely on that.

But if one of you would like to add support, we'd be happy for the contribution!

**Comment by MalteWunsch** [comment]
@tacman You're right, there is no release yet 😱 This is because we would have liked to tag a `1.0` as soon as it goes into production. And unfortunately, our only client (so far) with this bundle hasn't launched their website in over a year (reasons unrelated to this bundle). We've found other candidates for this bundle, but haven't implemented them yet.

Personally, I'd be fine to tag a `1.0` and accept a PR for Symfony 6 support, which may result in a `2.0` of this bundle.
